### PR TITLE
Uniformly apply toLowerCase for Dataflow Job Names

### PR DESCRIPTION
**Pull request by tgroh**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [x] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [x] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [x] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---
Also do minor cleanup of DataflowPIpelineOptions.

### CODE REVIEW

**Comment by tgroh** [comment]
R: @davorbonaci

File: runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/sdk/options/DataflowPipelineOptions.java [hunk 1]
```diff
@@ -68,7 +68,9 @@
    */
   @Description("The Dataflow job name is used as an idempotence key within the Dataflow service. "
       + "If there is an existing job that is currently active, another active job with the same "
-      + "name will not be able to be created. Defaults to using the ApplicationName-UserName-Date.")
+      + "name will not be able to be created. Defaults to using the ApplicationName-UserName-Date. "
```
Comment (davorbonaci) [inline runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/sdk/options/DataflowPipelineOptions.java]: If there's an existing and running job with the same name in the GCP project, you won't be able to start a new one unless the new job is an explicit update of the previous one.

File: runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/sdk/runners/DataflowPipelineRunner.java [hunk 2]
```diff
@@ -286,13 +286,26 @@ public static DataflowPipelineRunner fromOptions(PipelineOptions options) {
       LOG.debug("Classpath elements: {}", dataflowOptions.getFilesToStage());
     }
 
-    // Verify jobName according to service requirements.
-    String jobName = dataflowOptions.getJobName().toLowerCase();
-    Preconditions.checkArgument(
+    // Verify jobName according to service requirements, truncating converting to lowercase if
+    // necessary
```
Comment (davorbonaci) [inline runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/sdk/runners/DataflowPipelineRunner.java]: '.' at the end of the sentence.

File: runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/sdk/runners/DataflowPipelineRunner.java [hunk 3]
```diff
@@ -286,13 +286,26 @@ public static DataflowPipelineRunner fromOptions(PipelineOptions options) {
       LOG.debug("Classpath elements: {}", dataflowOptions.getFilesToStage());
     }
 
-    // Verify jobName according to service requirements.
-    String jobName = dataflowOptions.getJobName().toLowerCase();
-    Preconditions.checkArgument(
+    // Verify jobName according to service requirements, truncating converting to lowercase if
```
Comment (davorbonaci) [inline runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/sdk/runners/DataflowPipelineRunner.java]: the service requirements

File: runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/sdk/runners/DataflowPipelineRunner.java [hunk 4]
```diff
@@ -286,13 +286,26 @@ public static DataflowPipelineRunner fromOptions(PipelineOptions options) {
...
+            .toLowerCase();
+    checkArgument(
         jobName.matches("[a-z]([-a-z0-9]*[a-z0-9])?"),
         "JobName invalid; the name must consist of only the characters "
             + "[-a-z0-9], starting with a letter and ending with a letter "
             + "or number");
+    if (!jobName.equals(dataflowOptions.getJobName())) {
+      LOG.info(
+          "PipelineOptions.jobName did not match service requirements. "
+              + "Replacing original {} with {}.",
```
Comment (davorbonaci) [inline runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/sdk/runners/DataflowPipelineRunner.java]: Using X instead of Y.

**Comment by davorbonaci** [comment]
LGTM, modulo pending comments.

File: runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/sdk/options/DataflowPipelineOptions.java (hunk 1 above)
Comment (bjchambers) [inline runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/sdk/options/DataflowPipelineOptions.java]: Maybe "If there's an existing job running with the ..." or "If there's a running with the same name ...". The "an existing and running job" seems a bit odd.

File: runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/sdk/runners/DataflowPipelineRunner.java (hunk 2 above)
Comment (tgroh) [inline runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/sdk/runners/DataflowPipelineRunner.java]: Done

File: runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/sdk/options/DataflowPipelineOptions.java (hunk 1 above)
Comment (tgroh) [inline runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/sdk/options/DataflowPipelineOptions.java]: Reworded. PTAL

File: runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/sdk/runners/DataflowPipelineRunner.java (hunk 3 above)
Comment (tgroh) [inline runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/sdk/runners/DataflowPipelineRunner.java]: Done

File: runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/sdk/runners/DataflowPipelineRunner.java (hunk 4 above)
Comment (tgroh) [inline runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/sdk/runners/DataflowPipelineRunner.java]: Done

**Comment by davorbonaci** [comment]
Merge conflicts with the underlying branch. Please rebase.

**Comment by tgroh** [comment]
Done.

**Pull request merged**